src/adagrams.js

```diff
@@ -1,8 +1,144 @@
+// import { forEach } from "core-js/fn/array";
+
+const letterDist = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 1,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+};
+
 const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
+
+    // wave 1 - Draw a hand!
+
+    fillArray() {
+        const allLetters = []
+        for (const [key, value] of Object.entries(letterDist)) { // for all keys and values
+            for (let i = 0; i < value; i++) { // value number of times
+                allLetters.push(key); // push the letter into the allLetters array.
+            }
+        }
+        return allLetters;
+    },
+
+    shuffleArray() {
+        const array = Adagrams.fillArray()
+        for (let i = array.length - 1; i > 0; i--) { // array length times
+            let j = Math.floor(Math.random() * (i + 1)); // find random index
+            [array[i], array[j]] = [array[j], array[i]]; // swap last index with random index
+        }
+        return array;
+    },
+
+    drawLetters() {
+        const array = Adagrams.shuffleArray()
+        let letterHand = []
+        for (let i = 0; i < 10; i += 1) { // 10 times
+            letterHand.push(array.pop()); // take the last element from shuffled array
+        }
+        return letterHand;
+    },
+
+    // wave 2 -- Play from available letters! 
+
+    usesAvailableLetters(input, lettersInHand) {
+        if (input.length > lettersInHand.length) {
+            return false;
+        }
+
+        for (let char of input) {
+            if (!lettersInHand.includes(char)) {
+                return false;
+            }
+            lettersInHand.splice(lettersInHand.indexOf(char), 1);
+        }
+        return true;
+    },
+
+    // wave 3 -- Score word!
+
+    scoreWord(word) {
+        let score = 0
+
+        if (word.length > 6) {
+            score += 8;
+        };
+
+        for (let char of word) {
```

Comment:
This works and is a fun use of regex. Consider how you could DRY this up a bit if you stored the letter values in an object with the letter as the key, and score the value.